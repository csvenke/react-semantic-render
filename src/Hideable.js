import React from 'react'

/**
 * Higher order component that injects 'hide' prop into specified component.
 */
function Hideable (WrappedComponent) {
  return ({ hide, ...restProps }) =>
    !hide ? <WrappedComponent {...restProps} /> : null
}

export default Hideable
