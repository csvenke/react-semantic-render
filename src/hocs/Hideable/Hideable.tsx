import * as React from 'react';

import { renderIf } from '../../utils';

interface IHideableProps {
  hideComponent: boolean;
}

/**
 * Higher order component that injects 'hideComponent' prop into specified component.
 */
const Hideable = <P extends object>(
  Component: React.ComponentType<P>,
): React.SFC<P & IHideableProps> => ({ hideComponent, ...props }: IHideableProps) =>
  renderIf(!hideComponent)(<Component {...props} />);

export default Hideable;
