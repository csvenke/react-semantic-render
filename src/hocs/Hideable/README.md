### Example usage

```jsx
const Button = ({ label }) => <button>{label}</button>;
const HideableButton = Hideable(Button);

const App = ({ hide }) => <HideableButton hideComponent={hide} label="Click me!" />;

<App hide={false} />;
```
