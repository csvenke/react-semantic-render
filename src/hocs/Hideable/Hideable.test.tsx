import * as React from 'react';
import renderer from 'react-test-renderer';

import Hideable from '.';

const Button = ({ label }) => <button>{label}</button>;
const HideableButton = Hideable(Button);

test('should show component', () => {
  const component = renderer.create(
    <HideableButton hideComponent={false} label="Click me!" />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should return null', () => {
  const component = renderer.create(
    <HideableButton hideComponent={true} label="Click me!" />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
