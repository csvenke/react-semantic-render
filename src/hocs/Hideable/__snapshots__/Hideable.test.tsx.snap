// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return null 1`] = `null`;

exports[`should show component 1`] = `
<button>
  Click me!
</button>
`;
